tests/TopDownProteomics.Tests/ProForma/ChemicalProteoformHashTests.cs

```diff
@@ -0,0 +1,154 @@
+﻿using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using TopDownProteomics.Biochemistry;
+using TopDownProteomics.Chemistry;
+using TopDownProteomics.IO.Resid;
+using TopDownProteomics.ProForma;
+using TopDownProteomics.ProForma.Validation;
+using TopDownProteomics.Proteomics;
+using TopDownProteomics.Tests.IO;
+
+namespace TopDownProteomics.Tests.ProForma
+{
+    [TestFixture]
+    class ChemicalProteoformHashTests
+    {
+        IElementProvider _elementProvider;
+        IResidueProvider _residueProvider;
+        IProteoformModificationLookup _lookup;
+        IProteoformModification _acetyl;
+        string _acetylDescriptorString = "[formula:C2H2O1]";
+
+        [OneTimeSetUp]
+        public void Setup()
+        {
+            _elementProvider = new MockElementProvider();
+            _residueProvider = new IupacAminoAcidProvider(_elementProvider);
+
+            _lookup = new BrnoModificationLookup(_elementProvider);
+            _acetyl = _lookup.GetModification(new ProFormaDescriptor("ac(BRNO)"));
+        }
+
```

Comment:
You should add some sanity checking tests (probably just a single method to handle multiple cases). Right now this will all fail if you pass in NULL for example. What if the Modifications list on the group is NULL? etc.